### PR TITLE
Fixes /availability/single-view - start date doesn't reflects settings

### LINKED ISSUE #4380: /availability/single-view - start date doesn't reflects settings
Currently it always seems to start on a Sunday even if my week start date is Monday.

If there's going to be a default I would argue that it should be Monday. Especially since in our onboarding we have it as unavailable by default. This implies Monday is the logical start date and would make much more sense with our copy to all button.

CleanShot 2022-09-12 at 10 59 35@2x

### PR DESCRIPTION
**Pull request by emrysal**
Fixes #4380

### CODE REVIEW

File: apps/web/pages/v2/availability/[schedule].tsx [hunk 1]
```diff
@@ -132,7 +134,17 @@ export default function Availability({ schedule }: { schedule: number }) {
...
                 </h3>
-                <Schedule control={control} name="schedule" />
+                {typeof me.data?.weekStart === "string" && (
+                  <Schedule
+                    control={control}
+                    name="schedule"
+                    weekStart={
+                      ["Sunday", "Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"].indexOf(
+                        me.data?.weekStart
+                      ) as 0 | 1 | 2 | 3 | 4 | 5 | 6
```
Comment (hariombalhara) [inline apps/web/pages/v2/availability/[schedule].tsx:144]:
```suggestion
                      )
```

Comment (hariombalhara) [reply apps/web/pages/v2/availability/[schedule].tsx:144]: weekStart prop is a `number` type and thus it doesn't need type assertion
Also, the result can be -1 as well so assertion isn't correct.

**Review by hariombalhara — APPROVED** [review]

File: apps/web/pages/v2/availability/[schedule].tsx (hunk 1 above)
Comment (emrysal) [reply apps/web/pages/v2/availability/[schedule].tsx:144]: This is fair, it's a copy paste from similar places (I'm debating with myself whether to move weekStart logic into the schedule component itself).

Comment (hariombalhara) [reply apps/web/pages/v2/availability/[schedule].tsx:144]: maybe we can have MySchedule Higher Order Component that can have user specific settings preconfigured.

**Pull request merged**